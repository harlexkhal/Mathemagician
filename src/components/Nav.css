.nav-area {
  display: flex;
  width: 100%;
  height: 100px;
  background-color: purple;
  padding-bottom: 10px;
  margin-bottom: 50px;
}

.app-name-logo {
  font-size: xx-large;
  font-family: 'Comic Sans MS', serif;
  font-weight: 500;
  color: white;
  cursor: pointer;
}

.menu-container {
  margin: 25px 100px 0 auto;
}

.logo-container {
  display: flex;
  text-decoration: none;
}

.menu {
  display: flex;
  list-style: none;
}

.menu li {
  margin-right: 70px;
}

.menu-option-style {
  font-size: x-large;
  font-family: 'Comic Sans MS', serif;
  font-weight: 300;
  color: white;
  text-decoration: none;
  padding: 10px;
}

.menu-option-style:hover {
  color: rgb(255, 72, 255);
}

.menu-option-style:active {
  color: rgb(145, 73, 145);
}

.menu-option-style:focus {
  background-color: rgb(78, 8, 78);
}
