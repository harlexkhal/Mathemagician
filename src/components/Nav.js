import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../assets/logo.gif';
import './Nav.css';

class Nav extends React.PureComponent {
  render() {
    return (
      <nav className="nav-area">
        <Link className="logo-container" to="/">
          <img src={logo} className="App-logo" alt="logo" />
          <h3 className="app-name-logo">Mathemagician</h3>
        </Link>
        <div className="menu-container">
          <ul className="menu">
            <li><Link to="/" className="menu-option-style">Home</Link></li>
            <li><Link to="/calculator" className="menu-option-style">Calculator</Link></li>
            <li><Link to="/quotes" className="menu-option-style">Quotes</Link></li>
          </ul>
        </div>
      </nav>
    );
  }
}
export default Nav;
