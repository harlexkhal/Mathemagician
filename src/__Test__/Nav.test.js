import { render, screen, fireEvent } from '@testing-library/react';
import { HashRouter } from 'react-router-dom';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import Nav from '../components/Nav';
import '@testing-library/jest-dom';

expect.extend({ toMatchDiffSnapshot });

test('Make sure events does not make differences in snapshot (Nav Component) | [Home] Menu Click  Event Trigger', () => {
  const { asFragment } = render(<HashRouter><Nav /></HashRouter>);
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Home/));

  expect(firstRender).toMatchDiffSnapshot(asFragment());
});

test('Make sure events does not make differences in snapshot (Nav Component) | [Calculator] Menu Click  Event Trigger', () => {
  const { asFragment } = render(<HashRouter><Nav /></HashRouter>);
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Calculator/));

  expect(firstRender).toMatchDiffSnapshot(asFragment());
});

test('Make sure events does not make differences in snapshot (Nav Component) | [Quotes] Menu Click  Event Trigger', () => {
  const { asFragment } = render(<HashRouter><Nav /></HashRouter>);
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Quotes/));

  expect(firstRender).toMatchDiffSnapshot(asFragment());
});
