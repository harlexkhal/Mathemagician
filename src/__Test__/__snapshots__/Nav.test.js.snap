// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make sure events does not make differences in snapshot (Nav Component) | [Calculator] Menu Click  Event Trigger 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Make sure events does not make differences in snapshot (Nav Component) | [Home] Menu Click  Event Trigger 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Make sure events does not make differences in snapshot (Nav Component) | [Quotes] Menu Click  Event Trigger 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Make sure events does not make differences in snapshot (Nav Component) | Home Menu Click  Event Trigger 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
